### DISCUSSION TITLE
Hue distance calculation

### DISCUSSION THREAD

**Discussion by Cxom** [KyoriPowered/adventure#377] [General]
api/src/main/java/net/kyori/adventure/text/format/NamedTextColor.java Line 231:

`final float hueDistance = 3 * Math.abs(self.h() - other.h());`

Since hue is circular, should this not instead be the following?

`final float hueDistance = 3 * Math.min(Math.abs(self.h() - other.h()), 1f-Math.abs(self.h() - other.h()));`

**Comment by kashike** [comment]
@jpenilla You implemented the HSV code (#289) - thoughts?

**Reply by jpenilla** [reply]
Math checks out. I also tested this change in game, and you can see how it fixes the anti-red bias (as red is 360/0)

with this change: 
![](https://i.imgur.com/vymNp6v.png)

without this change(current impl): 
![](https://i.imgur.com/CVuwlaZ.png)

**Comment by kashike** [comment]
Thanks @Cxom - https://github.com/KyoriPowered/adventure/commit/10b7fd1eebce98d0ab4e04aa927bd9d9ad28e8aa
